android/src/com/frostwire/android/gui/activities/internal/MainController.java

```diff
@@ -28,46 +28,81 @@
 import com.frostwire.android.gui.fragments.TransfersFragment;
 import com.frostwire.android.gui.fragments.TransfersFragment.TransferStatus;
 import com.frostwire.android.gui.services.Engine;
+import com.frostwire.util.Ref;
+
+import java.lang.ref.WeakReference;
 
 /**
  * @author gubatron
  * @author aldenml
  */
 public final class MainController {
 
-    private final MainActivity activity;
+    private final WeakReference<MainActivity> activityRef;
 
     public MainController(MainActivity activity) {
-        this.activity = activity;
+        activityRef = Ref.weak(activity);
     }
 
     public MainActivity getActivity() {
-        return activity;
+        return activityRef.get();
     }
 
     public void closeSlideMenu() {
+        if (!Ref.alive(activityRef)) {
+            return;
+        }
+        MainActivity activity = activityRef.get();
         activity.closeSlideMenu();
     }
 
     public void switchFragment(int itemId) {
+        if (!Ref.alive(activityRef)) {
+            return;
+        }
+        MainActivity activity = activityRef.get();
         Fragment fragment = activity.getFragmentByNavMenuId(itemId);
         if (fragment != null) {
             activity.switchContent(fragment);
         }
     }
 
     public void showPreferences() {
+        if (!Ref.alive(activityRef)) {
+            return;
+        }
+        MainActivity activity = activityRef.get();
         Intent i = new Intent(activity, SettingsActivity.class);
         activity.startActivity(i);
     }
 
     public void launchMyMusic() {
+        if (!Ref.alive(activityRef)) {
+            return;
+        }
+        MainActivity activity = activityRef.get();
         Intent i = new Intent(activity, com.andrew.apollo.ui.activities.HomeActivity.class);
         i.setFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP | Intent.FLAG_ACTIVITY_NEW_TASK);
         activity.startActivity(i);
     }
 
+    public void launchPlayerActivity() {
```

Comment:
just moved this guy up, next to `launchMyMusic()`, one of the two appears to be unused, have to see if there's redundant code elsewhere, if not, it's going away after the merge.